### PR TITLE
Refactor creating Drivers

### PR DESCRIPTION
**Pull request by elahehrashedi**
the tests hasn't been fixed yet

### CODE REVIEW

File: src/cmakeProject.ts [hunk 1]
```diff
@@ -1157,6 +1123,25 @@ export class CMakeProject {
...
+        preconditionHandler: CMakePreconditionProblemSolver | undefined){
+            const workspaceContext = new DirectoryContext(workspaceFolder, config, new StateManager(new DefaultExtensionContext(), workspaceFolder));
+            const inst = new CMakeProject(workspaceContext, false);
+            await inst.init(workspaceFolder.uri.fsPath);
+            inst.setUseCMakePresets(false);
+            inst.setConfigurePreset(null);
+            inst.setBuildPreset(null);
+            inst.setTestPreset(null);
+            inst.setKit(kit);
+            if (!preconditionHandler){
```
Comment (elahehrashedi) [inline src/cmakeProject.ts:1139]: @bobbrow how can I pass the preconditionHandler to the project instance?

**Review by nasinlock — DISMISSED** [review]

File: src/cmakeProject.ts [hunk 2]
```diff
@@ -45,6 +45,8 @@ import { Environment, EnvironmentUtils } from './environmentVariables';
 import { KitsController } from './kitsController';
 import { PresetsController } from './presetsController';
 import paths from './paths';
+import { StateManager } from './state';
+import { DefaultExtensionContext } from '../test/helpers/vscodefake/extensioncontext';
```
Comment (bobbrow) [inline src/cmakeProject.ts:49]: This should not be imported here

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: src/drivers/cmakeDriver.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import * as vscode from 'vscode';
 
 import { CMakeExecutable } from '@cmt/cmake/cmakeExecutable';
 import * as codepages from '@cmt/codePageTable';
-import { ConfigureTrigger, DiagnosticsConfiguration } from "@cmt/cmakeProject";
+import CMakeProject, { ConfigureTrigger, DiagnosticsConfiguration } from "@cmt/cmakeProject";
```
Comment (bobbrow) [inline src/drivers/cmakeDriver.ts:10]: Move CMakeProject inside the curly braces

**Comment by elahehrashedi** [comment]
@bobbrow would you please check why the tests are failing? increasing the timeout doesn't help the test to pass.

**Comment by gcampbell-msft** [comment]
@benmcmorran @snehara99 Do we still want to eventually bring this up to date and merge this? I'm not sure exactly what it gains us, not knowing what the context was when it was created.

**Pull request closed without merging**